src/components/forms/MultiValueInput/MultiValueInput.tsx

```diff
@@ -236,6 +237,7 @@ const MultiValueInput: React.FC<MultiValueInputProps> = ({
     const pastedValue = (e.clipboardData || window.clipboardData).getData(
       'text',
     );
+    onPaste(e);
     addValue(pastedValue);
```

Comment:
Suggestion: check for e.defaultPrevented before calling addValue(). Would allow overriding the default behavior (for example filter our invalid values on custom validation)

Reply:
```
onPaste(e, pastedValue);
if (!e.defaultPrevented) {
    addValue(pastedValue);
}

```
This would allow this scenario:

```
onPaste={(e, pastedValue) => {
    e.preventDefault();
    parseValuesAndHandleInvalidDataExternally(pastedValue);
}
```

Reply:
Ah, sorry, didn’t see e.preventDefault() was already called above.

Alternative pattern :

```
if (onPaste(e, pastedValue) !== false) {
   addValue(pastedData)
}

having the callback explicitely return false would prevent the values from being added automatically.

```
onPaste={(e, pastedValue) => {
    parseValuesAndHandleInvalidDataExternally(pastedValue);
    return false; // Prevents default behavior
}
```
```